tw-entrypoints/src/main/java/com/transferwise/common/entrypoints/tableaccessstatistics/TableAccessStatisticsSpyqlListener.java

```diff
@@ -120,18 +127,26 @@ protected ParsedQuery parseSql(String sql, TwContext context) {
       return new ParsedQuery();
     }
 
-    ParsedQuery result = new ParsedQuery();
-    long startTimeMs = System.currentTimeMillis();
+    var result = new ParsedQuery();
+    var startTimeMs = System.currentTimeMillis();
     try {
       var stmts = sqlParser.parse(sql, entryPointsProperties.getTas().getSqlParser().getTimeout());
 
-      for (Statement stmt : stmts.getStatements()) {
+      for (var stmt : stmts) {
+        if (stmt instanceof UnsupportedStatement) {
+          throw new IllegalStateException("Unsupported statement.");
+        }
+      }
+
+      for (Statement stmt : stmts) {
```

Comment:
Line 135:
`for (var stmt : stmts) {`
Line 141:
`for (Statement stmt : stmts) {`

?